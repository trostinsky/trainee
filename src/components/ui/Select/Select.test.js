import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {mount} from "enzyme";
import Select from "./";
import 'jest-styled-components';

import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('Select match snapshot', () => {
    const select = renderer.create(
        <Select placeholder="Select" selectedIndex={1}>
            <Select.Option>1</Select.Option>
            <Select.Option>2</Select.Option>
            <Select.Option>4</Select.Option>
        </Select>
    );
    let tree = select.toJSON();
    expect(tree).toMatchSnapshot();
    let wrapper = mount(<Select placeholder="Select" selectedIndex={1}>
        <Select.Option>1</Select.Option>
        <Select.Option>2</Select.Option>
        <Select.Option>4</Select.Option>
    </Select>);
    wrapper.find('Select').first().simulate('click');


})