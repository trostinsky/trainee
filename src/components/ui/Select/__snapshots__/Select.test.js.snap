// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select match snapshot 1`] = `
.c5 {
  width: 100%;
  height: 60px;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 20px;
  margin-bottom: 5px;
  box-sizing: border-box;
  box-shadow: 9px 14px 36px -9px rgba(0,0,0,0.75);
}

.c2 {
  width: 20px;
  height: 20px;
  margin: 5px;
  padding: 0;
}

.c0 {
  position: relative;
  font-size: 13px;
  font-family: Chivo,sans-serif;
}

.c1 {
  -webkit-letter-spacing: 0.04em;
  -moz-letter-spacing: 0.04em;
  -ms-letter-spacing: 0.04em;
  letter-spacing: 0.04em;
  width: 100%;
  height: 60px;
  border: none;
  margin-bottom: 5px;
  max-width: 480px;
  background-color: #ffffff;
  font-family: Chivo,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 10px;
  box-sizing: border-box;
}

.c3 {
  margin: 0 auto;
  text-align: left;
  width: 100%;
  color: #333;
  padding: 0 9px;
}

.c4 {
  display: none;
  position: absolute;
  width: 80%;
  top: 50px;
  left: 50%;
  -webkit-transform: translate(-50%,0);
  -ms-transform: translate(-50%,0);
  transform: translate(-50%,0);
  color: #333;
}

<div
  className="c0"
>
  <div
    className=""
    onClick={[Function]}
  >
    <div
      className="c1"
    >
      <img
        className="c2"
        src="icon-username.svg"
      />
      <div
        className="c3"
      >
        2
      </div>
      <img
        className="c2"
        src="icon-username.svg"
      />
    </div>
  </div>
  <div
    className="c4"
  >
    <div
      onClick={[Function]}
    >
      <div
        className="c5"
      >
        1
      </div>
    </div>
    <div
      onClick={[Function]}
    >
      <div
        className="c5"
      >
        2
      </div>
    </div>
    <div
      onClick={[Function]}
    >
      <div
        className="c5"
      >
        4
      </div>
    </div>
  </div>
</div>
`;
